src/scriptdev2/scripts/eastern_kingdoms/sunken_temple/boss_atalai_defender.cpp

```diff
@@ -0,0 +1,251 @@
+/* Copyright (C) 2006 - 2011 ScriptDev2 <https://scriptdev2.svn.sourceforge.net/>
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software
+* Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+*/
+
+/* ScriptData
+SDName: Atal'ai Defenders
+SD%Complete: 90
+SDComment: Gasher, Hukku, Loro, Mijan, Zolo, Zul'Lor
+SDCategory: Sunken Temple
+EndScriptData */
+
+#include "precompiled.h"
+#include "sunken_temple.h"
+
+enum Spells
+{
+	// Gasher
+	SPELL_TRASH = 3391,
+	// Hukku
+	SPELL_CURSE_OF_BLOOD = 12279,
+	SPELL_HUKKUS_GUARDIANS = 12790,
+	SPELL_SHADOW_BOLT = 12471,
+	SPELL_SHADOW_BOLT_VOLLEY = 9081,
+	// Loro
+	SPELL_IMPROVED_BLOCKING = 3639,
+	SPELL_SHIELD_SLAM = 8242,
+	// Mijan
+	SPELL_HEALING_WARD = 11899,
+	SPELL_HEALING_WAVE = 12492,
+	SPELL_RENEW = 8362,
+	SPELL_THORNS_AURA = 8148,
+	// Zolo
+	SPELL_ATALAI_SKELETON_TOTEM = 12506,
+	SPELL_CHAIN_LIGHTNING = 12058,
+	// Zul'Lor
+	SPELL_CLEAVE = 15496,
+	SPELL_FRAILTY = 12530
+};
+
+struct boss_atalai_defenderAI : public ScriptedAI
+{
+	boss_atalai_defenderAI(Creature* pCreature) : ScriptedAI(pCreature)
+	{
+		m_pInstance = (instance_sunken_temple*)pCreature->GetInstanceData();
```

Comment:
If the code is good and we decide to do something with it I'll fix it up real fine ;)